File: examples/lib/src/main/java/com/github/neboskreb/log4j2/examples/lib/AwesomeWorker.java
package com.github.neboskreb.log4j2.examples.lib;

import lombok.extern.slf4j.XSlf4j;

@XSlf4j
public class AwesomeWorker {
    public String doGreatJob(String param1, String param2) {
        log.entry(param1, param2);

        String result = param1 + param2;

        return log.exit(result);
    }

    public void doNotSoGreatJob() {
        try {
            epicFail();

        } catch (Exception e) {
            log.warn("Arhhhh, the job would be great if not this epic failure...", e);
        }
    }

    public void epicFail() throws Exception {
        throw log.throwing(new Exception("Boo!"));
    }
}

File: examples/lib/src/main/kotlin/com/github/neboskreb/log4j2/examples/lib/AwesomeWorker.kt
package com.github.neboskreb.log4j2.examples.lib

import org.slf4j.ext.XLoggerFactory

private val log = XLoggerFactory.getXLogger(AwesomeWorker::class.java)

fun logSomething() {
    log.warn("A test warning from static context")
}

class AwesomeWorker {
    fun doGreatJob(param1: String, param2: String): String {
        log.entry(param1, param2)

        val result = param1 + param2

        return log.exit(result)
    }

    fun doNotSoGreatJob() {
        try {
            epicFail()
        } catch (e: Exception) {
            log.warn("Arhhhh, the job would be great if not this epic failure...", e)
        }
    }

    @Throws(Exception::class)
    fun epicFail() {
        throw log.throwing(Exception("Boo!"))
    }
}
